homeassistant/components/weatherflow_cloud/__init__.py

```diff
@@ -2,33 +2,121 @@
+        WebsocketObservation, ListenStartMessage, WebsocketObservation
+    ]
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up WeatherFlowCloud from a config entry."""
 
-    data_coordinator = WeatherFlowCloudDataUpdateCoordinator(
+    LOGGER.debug("Initializing WeatherFlowCloudDataUpdateCoordinatorREST coordinator")
+
+    rest_api = WeatherFlowRestAPI(
+        api_token=entry.data[CONF_API_TOKEN], session=async_get_clientsession(hass)
+    )
+
+    stations: StationsResponseREST = await rest_api.async_get_stations()
+
+    # Define Rest Coordinator
+    rest_data_coordinator = WeatherFlowCloudUpdateCoordinatorREST(
+        hass=hass, rest_api=rest_api, stations=stations
+    )
+
+    # Initialize the stations
+    await rest_data_coordinator.async_config_entry_first_refresh()
+
+    # Construct Websocket Coordinators
+    LOGGER.debug(
+        "Initializing WeatherFlowCloudDataUpdateCoordinatorWebsocketWind coordinator"
+    )
+    websocket_device_ids = rest_data_coordinator.device_ids
+    # Build API once
+    websocket_api = WeatherFlowWebsocketAPI(
+        access_token=entry.data[CONF_API_TOKEN], device_ids=websocket_device_ids
+    )
+
+    websocket_observation_coordinator = WeatherFlowCloudDataCallbackCoordinator[
+        WebsocketObservation, WebsocketObservation, ListenStartMessage
+    ](
+        hass=hass,
+        rest_api=rest_api,
+        websocket_api=websocket_api,
+        stations=stations,
+        listen_request_type=ListenStartMessage,
+        event_type=EventType.OBSERVATION,
+    )
+
+    websocket_wind_coordinator = WeatherFlowCloudDataCallbackCoordinator[
+        EventDataRapidWind, EventDataRapidWind, RapidWindListenStartMessage
+    ](
         hass=hass,
-        api_token=entry.data[CONF_API_TOKEN],
+        stations=stations,
+        rest_api=rest_api,
+        websocket_api=websocket_api,
+        listen_request_type=RapidWindListenStartMessage,
+        event_type=EventType.RAPID_WIND,
     )
-    await data_coordinator.async_config_entry_first_refresh()
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = data_coordinator
+    # Run setup method.
+    await asyncio.gather(
+        websocket_wind_coordinator._async_setup(),  # noqa: SLF001,
+        websocket_observation_coordinator._async_setup(),  # noqa: SLF001
+    )
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = WeatherFlowCoordinators(
+        rest_data_coordinator,
+        websocket_wind_coordinator,
+        websocket_observation_coordinator,
+    )
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Unload a config entry."""
+
+    # Note to Self / Reviewer
+    # - do we need code here to stop listening on the websockets?
+    # I need to check if they stay open after removing the integration
+
```

Comment:
Like the comment says. In local testing it seems to leave the coordinator running if I remove the integration

Reply:
`entry.async_on_unload` might be what you need

Reply:
Got it figured out